supertokens_python/recipe/multifactorauth/multi_factor_auth_claim.py

```diff
@@ -0,0 +1,245 @@
+                    "one_of": factor_ids,
+                },
+            )
+        else:
+            return ClaimValidationResult(
+                is_valid=False,
+                reason={
+                    "reason": f"Some of the factors are not complete in the session: {', '.join(factor_ids)}",
+                    "all_of_in_any_order": factor_ids,
+                },
+            )
+
+
+class HasCompletedRequirementListForAuthSCV(SessionClaimValidator):
+    def __init__(
+        self,
+        id_: str,
+        claim: MultiFactorAuthClaimClass,
+        mfa_claim_validators: MultiFactorAuthClaimValidators,
+        refetch_time_on_false_in_seconds: int,
+        max_age_in_seconds: Optional[int],
+    ):
+        super().__init__(id_)
+        self.claim = claim
+        self.mfa_claim_validators = mfa_claim_validators
+        self.refetch_time_on_false_in_ms = refetch_time_on_false_in_seconds * 1000
+        self.max_age_in_sec = max_age_in_seconds
+
+    async def validate(
+        self, payload: JSONObject, user_context: Dict[str, Any]
+    ) -> ClaimValidationResult:
+        if not self.claim.key in payload:
+            return ClaimValidationResult(
+                is_valid=False, reason={"message": "Claim value not present in payload"}
+            )
+
+        claim_val: MFAClaimValue = payload[self.claim.key]
+
+        if not claim_val.v:
+            return ClaimValidationResult(
+                is_valid=claim_val.v,
+                reason={
+                    "message": "MFA requirement for auth is not satisfied",
+                }
+                if not claim_val.v
+                else None,
+            )
+
+        return ClaimValidationResult(is_valid=True)
+
+
+class MultiFactorAuthClaimValidators:
+    def __init__(self, claim: MultiFactorAuthClaimClass):
+        self.claim = claim
+
+    def has_completed_requirement_list(
+        self, requirement_list: MFARequirementList, claim_key: Optional[str] = None
+    ) -> SessionClaimValidator:
+        return HasCompletedRequirementListSCV(
+            id_=claim_key or self.claim.key,
+            claim=self.claim,
+            mfa_claim_validators=self,
+            refetch_time_on_false_in_seconds=0,
+            max_age_in_seconds=None,
+            requirement_list=requirement_list,
+        )
+
+    def has_completed_mfa_requirements_for_auth(
+        self, claim_key: Optional[str] = None
+    ) -> SessionClaimValidator:
+
+        return HasCompletedRequirementListForAuthSCV(
+            id_=claim_key or self.claim.key,
+            claim=self.claim,
+            mfa_claim_validators=self,
+            refetch_time_on_false_in_seconds=0,
+            max_age_in_seconds=None,
+        )
+
+
+class MultiFactorAuthClaimClass(SessionClaim[MFAClaimValue]):
+    def __init__(self, key: Optional[str] = None):
+        key = key or "st-mfa"
+
+        async def fetch_value(
+            recipe_user_id: str,
```

Comment:
Add todo to add first arg as user_id: str,

Reply:
Added